src/librbd/journal/Types.h

```diff
@@ -121,12 +121,13 @@ struct OpFinishEvent : public OpEventBase {
 
 struct SnapEventBase : public OpEventBase {
   std::string snap_name;
+  cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
Fair enough -- although I'm still not sure how cloning is going to work for groups since OpenStack allows you to remove images from the group and yet still create group clones from a group snapshot. Lots of holes in the design.